hera_cal/tests/test_abscal.py

```diff
@@ -131,10 +145,78 @@ def test_gains2calfits(self):
         if os.path.exists(cfname):
             os.remove(cfname)
 
+    def test_wiener(self):
+        # test smoothing
+        d = hc.abscal.wiener(self.data, window=(5, 15), noise=None, medfilt=True, medfilt_kernel=(1, 13))
+        nt.assert_equal(d[(24,37,'xx')].shape, (60, 64))
+        nt.assert_equal(d[(24,37,'xx')].dtype, np.complex)
+        # test w/ noise
+        d = hc.abscal.wiener(self.data, window=(5, 15), noise=0.1, medfilt=True, medfilt_kernel=(1, 13))
+        nt.assert_equal(d[(24,37,'xx')].shape, (60, 64))
+        # test w/o medfilt
+        d = hc.abscal.wiener(self.data, window=(5, 15), medfilt=False)
+        nt.assert_equal(d[(24,37,'xx')].shape, (60, 64))
+        # test as array
+        d = hc.abscal.wiener(self.data[(24,37, 'xx')], window=(5, 15), medfilt=False, array=True)
+        nt.assert_equal(d.shape, (60, 64))
+        nt.assert_equal(d.dtype, np.complex)
+
+    def test_Baseline(self):
+        bls = map(lambda k: hc.abscal.Baseline(self.antpos[k[1]] - self.antpos[k[0]], tol=2.0), self.data.keys())
+        bls_conj = map(lambda k: hc.abscal.Baseline(self.antpos[k[0]] - self.antpos[k[1]], tol=2.0), self.data.keys())
+        nt.assert_equal(bls[0], bls[0])
+        nt.assert_false(bls[0] == bls[1])
+        nt.assert_equal(bls[0] == bls_conj[0], 'conjugated')
+
+    def test_match_red_baselines(self):
+        model = copy.deepcopy(self.data)
+        model = odict([((k[0]+1, k[1]+1, k[2]), model[k]) for i,k in enumerate(model.keys())])
+        model_antpos = odict([(k+1, self.antpos[k]) for i,k in enumerate(self.antpos.keys())])
+        data = hc.abscal.match_red_baselines(self.data, self.antpos, model, model_antpos, tol=2.0, verbose=False)
+        nt.assert_equal(len(data.keys()), 9)
```

Comment:
I'd like to see some more tests here showing that specific visibilities line up when they should and other that shouldn't do not line up

Reply:
yup